File: leetcode/java/0171_excel_sheet_column_number.java
class Solution {
    public int titleToNumber(String columnTitle) {
        Map<Character, Integer> hashMap = new HashMap<Character, Integer>();
        hashMap.put('A', 1);
        hashMap.put('B', 2);
        hashMap.put('C', 3);
        hashMap.put('D', 4);
        hashMap.put('E', 5);
        hashMap.put('F', 6);
        hashMap.put('G', 7);
        hashMap.put('H', 8);
        hashMap.put('I', 9);
        hashMap.put('J', 10);
        hashMap.put('K', 11);
        hashMap.put('L', 12);
        hashMap.put('M', 13);
        hashMap.put('N', 14);
        hashMap.put('O', 15);
        hashMap.put('P', 16);
        hashMap.put('Q', 17);
        hashMap.put('R', 18);
        hashMap.put('S', 19);
        hashMap.put('T', 20);
        hashMap.put('U', 21);
        hashMap.put('V', 22);
        hashMap.put('W', 23);
        hashMap.put('X', 24);
        hashMap.put('Y', 25);
        hashMap.put('Z', 26);

        int res = 0;
        for (int i = columnTitle.length() - 1, j = 0; i >= 0; i--, j++) {
            res += hashMap.get(columnTitle.charAt(i)) * Math.pow(26, j);
        }

        return res;
    }
}

File: leetcode/cpp/0171_excel_sheet_column_number.cpp
class Solution {
public:
    int titleToNumber(string columnTitle) {
        map<char, int> letters = { {'A', 1}, {'B', 2}, {'C', 3}, {'D', 4}, {'E', 5}, {'F', 6}, {'G', 7}, {'H', 8}, {'I', 9},
        {'J', 10}, {'K', 11}, {'L', 12}, {'M', 13}, {'N', 14}, {'O', 15}, {'P', 16}, {'Q', 17}, {'R', 18}, {'S', 19}, {'T', 20},
        {'U', 21}, {'V', 22}, {'W', 23}, {'X', 24}, {'Y', 25}, {'Z', 26} };
        int ret = 0;

        for (int i = 0, currIndex = columnTitle.size() - 1; i < columnTitle.size(); i++) {
            auto letterNum = letters.find(columnTitle[i])->second;

            ret += letterNum * pow(26, currIndex);
            currIndex--;
        }

        return ret;
    }
};